src/client/executor.rs

```diff
@@ -252,7 +252,15 @@ impl Client {
             let command_body = if should_redact {
                 Document::new()
             } else {
-                cmd.body.clone()
+                // Append versioned API fields
+                let mut body = cmd.body.clone();
+                if let Some(ref server_api) = self.inner.options.server_api {
+                    if server_api.applies_to_command(cmd.name.as_str()) {
+                        server_api.append_to_command(&mut body);
+                    }
```

Comment:
With the alternative API in ServerApi:
```suggestion
                    server_api.append_to_command_body(cmd.name.as_str(), &mut body);
```